File: Ceaser Cipher/CipherMachine.js

function isUpperCase(letter){
 if (letter >= 'A' && letter <= 'Z')
    return true
  return false
}

function isLowerCase(letter){
 if (letter >= 'a' && letter <= 'z')
    return true
  return false
}


function cipherEncript(words, key){
    let password = ""
    for (var letter of words){
        if(isUpperCase(letter)){
            let cipher = "" + String.fromCharCode(65 + ((letter.charCodeAt() - 65) + key) % 26)
            password += cipher
        }  
        else if (isLowerCase(letter)){
            cipher = "" + String.fromCharCode(97 + ((letter.charCodeAt() - 97) + key) % 26)
            password += cipher
        }       
        else password += letter
    }
    
    return password

}

function cipherDecript(words, key){
    let password = ""
    for (var letter of words){
        if(isUpperCase(letter)){
            let cipher = "" + String.fromCharCode(65 + ((letter.charCodeAt() - 65) - key) % 26)
            password += cipher
        }  
        else if (isLowerCase(letter)){
            cipher = "" + String.fromCharCode(97 + ((letter.charCodeAt() - 97) - key) % 26)
            password += cipher
        }       
        else password += letter
    }
    
    return password

}

console.log(cipherEncript("Samuel..",3))
console.log(cipherDecript("Vdpxho..",3))

File: Ceaser Cipher/CipherMachine.py

def cipher_encript(words, key):
    password = ""
    for letter in words:
        if letter.isupper():
            cipher = "" + chr(65 + ((ord(letter) - 65) + key) % 26)
            password += cipher
        elif letter.islower():
            cipher = "" + chr(97 + ((ord(letter) - 97) + key) % 26)
            password += cipher
        else:
            password += letter
    return password


def cipher_decript(words, key):
    password = ""
    for letter in words:
        if letter.isupper():
            cipher = "" + chr(65 + ((ord(letter) - 65) - key) % 26)
            password += cipher
        elif letter.islower():
            cipher = "" + chr(97 + ((ord(letter) - 97) - key) % 26)
            password += cipher
        else:
            password += letter
    return password



print(cipher_encript("Samuel..", 3))
print(cipher_decript("Vdpxho..", 3))

